### PR TITLE
metrics: add some forwarding metrics

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

We don't have any metrics to show the status of forwarding requests.

### What is changed and how it works?

This PR adds some metrics to show it.

### Check List

Tests

- Manual test (TODO)

Related changes

- Need to cherry-pick to the release branch

### Release note

- Add some metrics to show the states of forwarding requests

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -363,6 +376,8 @@ func (s *Server) PutStore(ctx context.Context, request *pdpb.PutStoreRequest) (*
 func (s *Server) GetAllStores(ctx context.Context, request *pdpb.GetAllStoresRequest) (*pdpb.GetAllStoresResponse, error) {
 	forwardedHost := getForwardedHost(ctx)
 	if !s.isLocalRequest(forwardedHost) {
+		start := time.Now()
```
Comment (nolouch) [inline server/grpc_service.go:379]: I think grpc already has the duration metrics. Do we need to add the new?

**Review by nolouch — APPROVED** [review]
LGTM

File: client/metrics.go [hunk 2]
```diff
@@ -51,6 +51,13 @@ var (
 			Help:      "Bucketed histogram of the batch size of handled requests.",
 			Buckets:   prometheus.ExponentialBuckets(1, 2, 13),
 		})
+	requestForwarded = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "pd_client",
+			Subsystem: "request",
+			Name:      "forwarded_status",
+			Help:      "The status to indicate if the request is forwarded",
+		}, []string{"host", "delegate"})
```
Comment (Yisaer) [inline client/metrics.go:60]: What about add a label to indicate the proxy pd member name?

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**